manifests/init.pp

```diff
@@ -342,16 +322,6 @@
   Boolean $manage_puppet_group = $foreman_proxy::params::manage_puppet_group,
   Boolean $puppet = $foreman_proxy::params::puppet,
   Foreman_proxy::ListenOn $puppet_listen_on = $foreman_proxy::params::puppet_listen_on,
-  Optional[Enum['mcollective', 'ssh', 'salt', 'customrun']] $puppetrun_provider = $foreman_proxy::params::puppetrun_provider,
-  String $customrun_cmd = $foreman_proxy::params::customrun_cmd,
-  String $customrun_args = $foreman_proxy::params::customrun_args,
-  String $mcollective_user = $foreman_proxy::params::mcollective_user,
-  Boolean $puppetssh_sudo = $foreman_proxy::params::puppetssh_sudo,
-  String $puppetssh_command = $foreman_proxy::params::puppetssh_command,
-  String $puppetssh_user = $foreman_proxy::params::puppetssh_user,
-  Stdlib::Absolutepath $puppetssh_keyfile = $foreman_proxy::params::puppetssh_keyfile,
-  Boolean $puppetssh_wait = $foreman_proxy::params::puppetssh_wait,
-  String $salt_puppetrun_cmd = $foreman_proxy::params::salt_puppetrun_cmd,
   String $puppet_user = $foreman_proxy::params::puppet_user,
```

Comment:
This parameter should also be removed since you've removed it from params.pp. That causes all the tests to fail.